### PR TITLE
Edit Site: Refactor business logic into store.

### PR DESCRIPTION
**Pull request by epiqueras**
## Description

This PR refactors all of the `edit-site` business logic and editor state into the store.

It also moves a local utility, `getPathAndQueryString`, into the `@wordpress/url` package.

## How has this been tested?

Tests were written for all the new functions, and it was verified that nothing changed in the site editor.

## Types of Changes

*New Feature:* `@wordpress/url` now has a `getPathAndQueryString` utility.

## Checklist:

- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by Addison-Stavlo — COMMENTED** [review]
This is looking great and cleans things up quite a bit.  Thank you!

Reading through the code changes things look good.

From what I can tell mostly everything in the site editor works as expected in testing.  However, I notice that when I test the template switcher's "Overwrite Template" / "Revert to Parent" functionality for a page, the revert doesn't go through as on Master.  Eventually get a 410 console error with `Uncaught (in promise) {code: "rest_already_trashed", message: "The post has already been deleted.", data: {…}}`

**Comment by epiqueras** [comment]
Oops, I missed that part while refactoring. All done now 😄

**Review by Addison-Stavlo — APPROVED** [review]
Awesome, thanks for the quick fix there too!  

Ive gone through and tested everything I can think of again in the site editor, and it's all working as expected.  After combing through the code again, things look good to me!

File: packages/edit-site/src/store/selectors.js [hunk 1]
```diff
@@ -25,3 +32,137 @@ export function isFeatureActive( state, feature ) {
...
+/**
+ * Returns the current template IDs.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {number[]} Template IDs.
+ */
+export function getTemplateIds( state ) {
+	return state.templateIds;
+}
```
Comment (ockham) [inline packages/edit-site/src/store/selectors.js:135]: AFAICS, the `getTemplateIds` selector isn't used anywhere. We could consider dropping it (and the corresponding `templateIds` reducer, plus potentially some related actions, resolvers, and `editorInitialState`).

Comment (Addison-Stavlo) [reply packages/edit-site/src/store/selectors.js:135]: I think they might come in handy down the road though?  If I'm understanding it right their values are being initialized from the `edit-site-page.php`, so now getting them will be easily accessible in future development.

Comment (epiqueras) [reply packages/edit-site/src/store/selectors.js:135]: @ockham Worked on removing it from `edit-site` state and instead use the `getEntityRecords` API with a new query parameter.

I think that, in #21878, we should remove both the `templateIDs` and `templatePartIds` selectors and reducers, as well as the code in `edit-site-page` that loads it.

**Comment by youknowriad** [comment]
Nice to see this PR. Thanks @epiqueras

**Pull request merged**